MODULE.bazel

```diff
@@ -1,6 +1,50 @@
-###############################################################################
-# Bazel now uses Bzlmod by default to manage external dependencies.
-# Please consider migrating your external dependencies from WORKSPACE to MODULE.bazel.
-#
-# For more details, please check https://github.com/bazelbuild/bazel/issues/18958
-###############################################################################
+module(
+    name = "openroad",
+)
+
+bazel_dep(name = "platforms", version = "0.0.11")
+bazel_dep(name = "rules_bison", version = "0.3.1")
+bazel_dep(name = "rules_flex", version = "0.3.1")
+bazel_dep(name = "rules_m4", version = "0.2.3")
```

Comment:
why do we need m4?

Reply:
m4 is used by bison internally.

Reply:
Then wouldn't that be internal to bison and not a direct dependency?  I thought a goal for MODULE is to avoid having to mention transitive dependencies.

Reply:
It possibly does that when just using the rules provided by `rules_bison` ( https://github.com/jmillikin/rules_bison?tab=readme-ov-file#examples ). However here, while in the transition from bazel_rules_hdl I added `baze/bizon.bzl` to be compatible with the previously used `genlex()` and `genyacc()` rules, so need to explicitly refer to `m4`.

So after the dust settles, we can possibly use directly the rules provided in https://github.com/jmillikin/rules_bison/blob/trunk/docs/rules_bison.md

Reply:
Note, I also added a flag `--//bazel:use_local_flex_bison` that allows to use the locally installed flex/bison which allows an escape-hatch in case people run into issues (the gnulib needed for flex and bison used to have compile issues on MacOS and Windows).

The flag is defined in the `bazel/BUILD` file:

```python
bool_flag(
    name = "use_local_flex_bison",
    build_setting_default = False,
)

config_setting(
    name = "use_local_flex_bison_enabled",
    flag_values = {":use_local_flex_bison": "true"},
)
```

.. and then referenced in the bison.bzl and flex.bzl files to choose what to do.

Reply:
Make sense